agrest-cayenne/src/main/java/io/agrest/cayenne/exp/CayenneExpPostProcessor.java

```diff
@@ -215,4 +306,29 @@ private String findChildPath(Expression exp) {
             return null;
         }
     }
+
+    static class ObjPathMarker extends ASTObjPath {
+
+        final ObjRelationship relationship;
+
+        ObjPathMarker(String path, ObjRelationship relationship) {
+            super(path);
+            this.relationship = relationship;
+        }
+
+        @Override
```

Comment:
Need to check do we need any methods here. Those are needed in Cayenne to add this markers to collections.